pkg/virt-controller/services/template.go

```diff
@@ -992,8 +1010,13 @@ func getCniInterfaceList(vmi *v1.VirtualMachineInstance) (ifaceListString string
 			}
 		}
 	}
-
-	ifaceListString = strings.Join(ifaceList, ",")
+	if len(ifaceListMap) > 0 {
+		//ifaceListString, _ = json.Marshal(ifaceListMap)
+		ifaceJsonString, _ := json.Marshal(ifaceListMap)
+		ifaceListString = fmt.Sprintf("%s", ifaceJsonString)
+	} else {
+		ifaceListString = strings.Join(ifaceList, ",")
```

Comment:
What's the reason for us to interchange format of the annotation based on the length of the map-list? Wouldn't it work even if we use json format for a 1-list? If 1-list json still works, this seems to introduce unnecessary complexity.

Reply:
Reason is that I do not know if Genie can deal with JSON list. In case it can -> I agree with your statement and we should use JSON list here as well.

Reply:
Please add an explanation in the comment on why we keep this logic (about Genie).